sklearn/metrics/_plot/regression.py

```diff
@@ -0,0 +1,336 @@
+            line_kwargs = {}
+        if residuals_kwargs is None:
+            residuals_kwargs = {}
+
+        default_scatter_kwargs = {"color": "black", "alpha": 0.8}
+        default_line_kwargs = {
+            "color": "tab:blue",
+            "alpha": 0.7,
+            "linestyle": "--",
+        }
+        default_residuals_kwargs = {"color": "red", "alpha": 0.5}
+
+        scatter_kwargs = {**default_scatter_kwargs, **scatter_kwargs}
+        line_kwargs = {**default_line_kwargs, **line_kwargs}
+        residuals_kwargs = {**default_residuals_kwargs, **residuals_kwargs}
+
+        if ax is None:
+            _, ax = plt.subplots()
+
+        if self.with_residuals:
+            self.residual_lines_ = []
+            for actual, predicted in zip(self.y_true, self.y_pred):
+                residual_line = ax.plot(
+                    [actual, actual], [actual, predicted], **residuals_kwargs,
+                )
+                self.residual_lines_ += residual_line
+        else:
+            self.residual_lines_ = None
+
+        max_value = max(np.max(self.y_true), np.max(self.y_pred))
+        min_value = min(np.min(self.y_true), np.min(self.y_pred))
+        self.line_ = ax.plot(
+            [min_value, max_value], [min_value, max_value], **line_kwargs
+        )[0]
+
+        self.scatter_ = ax.scatter(self.y_true, self.y_pred, **scatter_kwargs)
+
+        xlabel, ylabel = "Actual values", "Predicted values"
+        ax.set(xlabel=xlabel, ylabel=ylabel)
+        ax.set_aspect('equal', adjustable='datalim')
+        ax.set_xticks(np.linspace(min_value, max_value, num=5))
+        ax.set_yticks(np.linspace(min_value, max_value, num=5))
+
+        if self.scores is not None:
+            extra = plt.Rectangle(
+                (0, 0), 0, 0, fc="w", fill=False, edgecolor="none", linewidth=0
+            )
+            scoring_legend = "\n".join(
+                [
+                    f"{name} = {value}"
+                    for name, value in self.scores.items()
+                ],
+            )
+            ax.legend([extra], [scoring_legend])
+
+        self.ax_ = ax
+        self.figure_ = ax.figure
+
+        return self
+
+
+def plot_prediction_error(
+    estimator,
+    X,
+    y,
+    *,
+    scoring=None,
+    subsample=None,
+    random_state=None,
+    with_residuals=False,
+    ax=None,
+    scatter_kwargs=None,
+    line_kwargs=None,
+    residuals_kwargs=None,
+):
+    """Plot the prediction error of a regressor.
+
+    Extra keyword arguments will be passed to matplotlib's ``plot``.
+
+    Read more in the :ref:`User Guide <visualizations>`.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Fitted regressor or a fitted :class:`~sklearn.pipeline.Pipeline`
+        in which the last estimator is a regressor.
+
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        Input values.
+
+    y : array-like of shape (n_samples,)
+        Target values.
```

Comment:
in relation to my previous comments, I'm wondering whether it really makes sense to pass X and y here. Maybe it should be `y` and `y_pred` instead: at least according to our examples, having `y_pred` already computed seems to be the most common use-case.

Reply:
Or maybe we should accept both strategies...

Some related discussion in https://github.com/scikit-learn/scikit-learn/issues/15880

Reply:
Yes, we encounter this issue where we would like to use precomputed prediction and score.
We actually seem to not have to figure out the API that we want to go with. I don't know if would be safer to restrict the current signature and make a move afterwards.

Reply:
CC @thomasjpfan @amueller 

One difference here vs #15880 is that we don't have to worry about backward compat, and we can come up with whatever API we want/need.


It seems to me that allowing both `estimator, X` and `y_pred` would be the most natural solution here (while keeping them mutually exclusive).

Reply:
Having a single first tuple argument with `(estimator, X, y)` or `(y_true, y_pred)` would be nice in this case.
You can switch easily by detecting in which case you are and might not be that user-unfriendly.